test/core/utils/respondable.js

```diff
@@ -1,515 +1,561 @@
+        receivedResponse = message;
       });
 
-      setTimeout(function() {
-        done();
-      }, 100);
+      afterMessage(
+        window,
+        captureError(function() {
+          assert.deepEqual(receivedResponse, { greet: 'bonjour' });
+          done();
+        }, done)
+      );
     });
 
-    it('should reject messages that do not have a matching uuid', function(done) {
-      eventData = {
-        _respondable: true,
-        _source: 'axeAPI.' + axe.version,
-        message: 'Help us Obi-Wan',
-        _axeuuid: 'otherAxe'
-      };
-
-      win = {
-        postMessage: function(message) {
-          var data = JSON.parse(message);
-          eventData.uuid = data.uuid + 'joker tricks!';
-          event.data = JSON.stringify(eventData);
-          document.dispatchEvent(event);
-        }
-      };
+    it('prohibits multiple response calls with keepalive: false', function(done) {
+      var receivedResponse;
+      frameSubscribe('greeting', function(message, keepalive, respond) {
+        assert.isTrue(keepalive);
+        respond({ responseNum: 1 }, false);
+        setTimeout(function() {
+          respond({ responseNum: 2 });
+        }, 10);
+      });
 
-      axe.utils.respondable(win, 'Death star', null, true, function() {
-        done(new Error('should not call callback'));
+      respondable(frameWin, 'greeting', '', true, function(message) {
+        receivedResponse = message;
       });
 
-      setTimeout(function() {
-        done();
-      }, 100);
+      afterMessage(window, function() {
+        assert.deepEqual(receivedResponse, { responseNum: 1 });
+        setTimeout(function() {
+          assert.deepEqual(receivedResponse, { responseNum: 1 });
+          done();
+        }, 100);
+      });
     });
 
-    it('should reject messages that do not have `_respondable: true`', function(done) {
-      eventData = {
-        _source: 'axeAPI.' + axe.version,
-        message: 'Help us Obi-Wan',
-        _axeuuid: 'otherAxe'
-      };
+    it('allows multiple response calls with keepalive: true', function(done) {
+      var receivedResponse;
+      frameSubscribe('greeting', function(message, keepalive, respond) {
+        assert.isTrue(keepalive);
+        respond({ responseNum: 1 }, true);
+        setTimeout(function() {
+          respond({ responseNum: 2 });
+        }, 30);
+      });
 
-      axe.utils.respondable(win, 'Death star', null, true, function() {
-        done(new Error('should not call callback'));
+      respondable(frameWin, 'greeting', '', true, function(message) {
+        receivedResponse = message;
       });
 
-      setTimeout(function() {
-        done();
-      }, 100);
+      afterMessage(window, function() {
+        assert.deepEqual(receivedResponse, { responseNum: 1 });
+        afterMessage(window, function() {
+          assert.deepEqual(receivedResponse, { responseNum: 2 });
+          done();
+        });
```

Comment:
I feel like this can easily lead to a timing problem without the longer timeout delay like the prior test

```suggestion
        }, 100);
```

Reply:
It won't, this runs after the second message is received. There is no race condition here.